frontend/pages/admin/IntegrationsPage/cards/ConditionalAccess/components/OktaConditionalAccessModal/OktaConditionalAccessModal.tsx

```diff
@@ -0,0 +1,396 @@
+  const errs: IFormErrors = {};
+
+  // Max lengths from backend validation
+  const maxURLLength = 2048;
+  const maxCertLength = 8192;
+
+  // IdP ID validation - must be non-empty and not just whitespace
+  if (!formData[OKTA_IDP_ID] || !formData[OKTA_IDP_ID].trim()) {
+    errs[OKTA_IDP_ID] = "IdP ID must be present";
+  } else if (formData[OKTA_IDP_ID].length > maxURLLength) {
+    errs[OKTA_IDP_ID] = `IdP ID must be ${maxURLLength} characters or less`;
+  }
+
+  // Assertion Consumer Service URL validation
+  if (!formData[OKTA_ACS_URL] || !formData[OKTA_ACS_URL].trim()) {
+    errs[OKTA_ACS_URL] = "Assertion Consumer Service URL must be present";
+  } else if (formData[OKTA_ACS_URL].length > maxURLLength) {
+    errs[
+      OKTA_ACS_URL
+    ] = `Assertion Consumer Service URL must be ${maxURLLength} characters or less`;
+  } else {
+    // Validate URL format - must have http or https scheme and a host
+    try {
+      const acsURL = new URL(formData[OKTA_ACS_URL]);
+      if (
+        (acsURL.protocol !== "http:" && acsURL.protocol !== "https:") ||
+        !acsURL.host
+      ) {
+        errs[OKTA_ACS_URL] =
+          "Assertion Consumer Service URL must be a valid URL with http or https scheme and a host";
+      }
+    } catch {
+      errs[OKTA_ACS_URL] =
+        "Assertion Consumer Service URL must be a valid URL with http or https scheme and a host";
+    }
+  }
+
+  // Audience URI validation
+  if (!formData[OKTA_AUDIENCE_URI] || !formData[OKTA_AUDIENCE_URI].trim()) {
+    errs[OKTA_AUDIENCE_URI] = "Audience URI must be present";
+  } else if (formData[OKTA_AUDIENCE_URI].length > maxURLLength) {
+    errs[
+      OKTA_AUDIENCE_URI
+    ] = `Audience URI must be ${maxURLLength} characters or less`;
+  }
+
+  // Certificate validation
+  if (!formData[OKTA_CERTIFICATE] || !formData[OKTA_CERTIFICATE].trim()) {
+    errs[OKTA_CERTIFICATE] = "Certificate must be present";
+  } else if (formData[OKTA_CERTIFICATE].length > maxCertLength) {
+    errs[
+      OKTA_CERTIFICATE
+    ] = `Certificate must be ${maxCertLength} characters or less`;
+  }
+
+  return errs;
+};
+
+export interface IOktaConditionalAccessModalProps {
+  onCancel: () => void;
+  onSuccess: (updatedConfig: IConfig) => void;
+}
+
+const OktaConditionalAccessModal = ({
+  onCancel,
+  onSuccess,
+}: IOktaConditionalAccessModalProps) => {
+  const { renderFlash } = useContext(NotificationContext);
+  const { config } = useContext(AppContext);
+
+  const [isUpdating, setIsUpdating] = useState(false);
+  const [formData, setFormData] = useState<IFormData>({
+    [OKTA_IDP_ID]: "",
+    [OKTA_ACS_URL]: "",
+    [OKTA_AUDIENCE_URI]: "",
+    [OKTA_CERTIFICATE]: "",
+  });
+  const [formErrors, setFormErrors] = useState<IFormErrors>({});
+  const [certFile, setCertFile] = useState<File | null>(null);
+  const [appleProfile, setAppleProfile] = useState<string>("");
+
+  // Fetch Apple profile on mount
+  useEffect(() => {
+    const fetchAppleProfile = async () => {
```

Comment:
Could you just put this logic outside of this function and useEffect? e.g.:
```
try {
  const profileText = await ...
  ...
} catch (e: any) {
  ...
renderFlash("error", message);
}
```

Reply:
Or use our typical `useQuery` pattern, and maybe have it give you a couple retries before erroring